### ISSUE TITLE
BUG: breakpoint causes valid code to have a syntax error

### ISSUE THREAD

**Issue by iBicha** [rokucommunity/roku-debug#111]

**Comment by TwitchBronBron** [comment]
Yeah, that makes sense why it doesn't work. I wonder if there is any way we could inject the stop statement safely in that situation?  Like, would something like:

``` 
params = { 
    stop; q: q
}```

Probably not though. 

We don't currently have any validation for where breakpoints are placed. However, we definitely should. Maybe parse the file with the brighterscript compiler and inspect the line to ensure it's going to be placing the breakpoint at a valid location?

**Comment by iBicha** [comment]
> Maybe parse the file with the brighterscript compiler and inspect the line to ensure it's going to be placing the breakpoint at a valid location?

Can't think of any other way tbh

**Comment by chrisdp** [comment]
We don't honestly do what Ts does and move it to the next valid stop inlet line. Like @iBicha says we might be able to do something where we use brighter script to parse and find that next valid location.

**Comment by iBicha** [comment]
> We should honestly do what Ts does and move it to the next valid stopable line. Like @iBicha says we might be able to do something where we use brighter script to parse and find that next valid location.

Actually it's not a lot of effort for the dev to move the breakpoint themselves. The problem is not knowing why it is failing. So if detection is easier than prevention, it's definitely a good path with just a good error message

**Comment by chrisdp** [comment]
@iBicha i think that what I was getting at is that detecting it is almost as much as moving it for them since we need to understand all invalid locations. But yea an error would help as well.
